lib/shell.js

```diff
@@ -34,20 +34,19 @@ const async = require('async'),
                     }
                 },
                 function(next) {
-                    const execOption = {};
                     if (options.sessionId) {
                         if (options.session.getDevice().username !== 'root') {
                             return setImmediate(next, errHndl.changeErrMsg("NEED_ROOT_PERMISSION", "connect user session"));
                         }
 
                         // -l : make the shell a login shell, -c : command
-                        runCommand = `su ${options.sessionId} -l -c "${runCommand}"`;
-                        execOption.pty = true;
+                        runCommand = `su ${options.sessionId} -l -c '${runCommand}'`;
+                    } else {
+                        runCommand = "source /etc/profile && " + runCommand;
```

Comment:
I have a question. 
I understood that you would use "su root" for root shell connection instead of "source /etc/profile". right?
However, your code keep "source /etc/profile". 
Did you determine use "source /etc/profile" for root shell?